### DISCUSSION TITLE
no gpu detection

### DISCUSSION THREAD

**Discussion by sskim5128** [marcelotduarte/cx_Freeze#2086] [Q&A]
**When running a python file in visual code, GPU is recognized as following :**

tensorflow : 2.10.0
2023-10-05 11:59:10.224974: W tensorflow/core/common_runtime/pluggable_device/pluggable_device_bfc_allocator.cc:28] Overriding allow_growth setting because force_memory_growth was requested by the 
device.
CPU/GPU : [name: "/device:CPU:0"
device_type: "CPU"
memory_limit: 268435456
locality {
}
incarnation: 7783995792683096882
xla_global_id: -1
, name: "/device:GPU:0"
device_type: "GPU"
memory_limit: 18688519092
locality {
  bus_id: 1
}
incarnation: 4622110639276052480
physical_device_desc: "device: 0, name: DML, pci bus id: <undefined>"
xla_global_id: -1
]
tensorflow build info : OrderedDict([('is_cuda_build', False), ('is_rocm_build', False), ('is_tensorrt_build', False), ('msvcp_dll_names', 'msvcp140.dll,msvcp140_1.dll')])

**But, the GPU is not recognized in the executable created with cx_freeze**
tensorflow : 2.10.0
physical devices : [PhysicalDevice(name='/physical_device:CPU:0', device_type='CPU')]
logical devices : [LogicalDevice(name='/device:CPU:0', device_type='CPU')]
CPU/GPU : [name: "/device:CPU:0"
device_type: "CPU"
memory_limit: 268435456
locality {
}
incarnation: 8583048764476748566
xla_global_id: -1
]
tensorflow build info : OrderedDict([('is_cuda_build', False), ('is_rocm_build', False), ('is_tensorrt_build', False), ('msvcp_dll_names', 'msvcp140.dll,msvcp140_1.dll')])
current runtime path : E:\vsc\python_projects\build\exe.win-amd64-3.10

**Comment by marcelotduarte** [comment]
Unfortunately I have no way of testing this. But when I was solving an issue related to tensorflow, I read in a project, which I no longer remember which one it was, that using cx_Freeze was the way with the best results for initialization, since tensorflow loads everything at the beginning. Maybe an earlier version of tensorflow.

**Reply by marcelotduarte** [reply]
Please check https://github.com/marcelotduarte/cx_Freeze/issues/2300#issuecomment-2000135037

**Comment by marcelotduarte** [comment]
[Release 6.15.16](https://github.com/marcelotduarte/cx_Freeze/releases/tag/6.15.16) is out!
